File: app/src/main/java/es/etologic/mahjongscoring2/app/base/BaseFragment.java
package es.etologic.mahjongscoring2.app.base;

import android.app.Activity;
import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.util.Log;
import android.view.View;

import javax.inject.Inject;

import butterknife.ButterKnife;
import butterknife.Unbinder;
import dagger.android.support.DaggerFragment;

public abstract class BaseFragment extends DaggerFragment {

    //FIELDS
    @Inject protected Context context;
    private Unbinder viewUnbinder;

    //LIFECYCLE
    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        viewUnbinder = ButterKnife.bind(this, view);
    }
    protected void showError(Throwable throwable) {
        Activity activity = getActivity();
        if(activity instanceof BaseActivity) {
            ((BaseActivity) activity).showError(throwable);
        }
    }
    protected void showSnackbar(View view, String message) {
        Activity activity = getActivity();
        if(activity instanceof BaseActivity) {
            ((BaseActivity) activity).showSnackbar(view, message);
        }
    }
    @Deprecated
    @Override
    public void startActivity(Intent intent) {
        Log.d("BaseFragment", "StartActivity is forbidden from Fagments");
    }
    @Deprecated
    @Override
    public void startActivity(Intent intent, @Nullable Bundle options) {
        Log.d("BaseFragment", "StartActivity is forbidden from Fagments");
    }
    @Override
    public void onDestroyView() {
        if(viewUnbinder != null) {
            viewUnbinder.unbind();
        }
        super.onDestroyView();
    }
}

File: app/src/main/java/es/etologic/mahjongscoring2/app/base/BaseFragment.kt
package es.etologic.mahjongscoring2.app.base

import android.content.Intent
import android.os.Bundle
import android.util.Log
import android.view.View
import dagger.android.support.DaggerFragment

abstract class BaseFragment : DaggerFragment() {
    
    protected fun showError(throwable: Throwable) {
        val activity = activity
        if (activity is BaseActivity) activity.showError(throwable)
    }
    
    protected fun showSnackbar(view: View, message: String) {
        val activity = activity
        if (activity is BaseActivity) activity.showSnackbar(view, message)
    }
    
    @Deprecated("StartActivity is forbidden from Fagments")
    override fun startActivity(intent: Intent) {
        Log.d("BaseFragment", "StartActivity is forbidden from Fagments")
    }
    
    @Deprecated("StartActivity is forbidden from Fagments")
    override fun startActivity(intent: Intent, options: Bundle?) {
        Log.d("BaseFragment", "StartActivity is forbidden from Fagments")
    }
}
